custom_components/tesla_custom/device_tracker.py

```diff
@@ -1,63 +1,79 @@
 """Support for tracking Tesla cars."""
-from __future__ import annotations
+import logging
 
 from homeassistant.components.device_tracker import SOURCE_TYPE_GPS
 from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+
+from . import TeslaDataUpdateCoordinator
+from .base import TeslaBaseEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entities):
+    """Set up the Tesla selects by config_entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+    for car in hass.data[DOMAIN][config_entry.entry_id]["cars"]:
+        entities.append(CarLocation(hass, car, coordinator))
 
-from . import DOMAIN as TESLA_DOMAIN
-from .tesla_device import TeslaDevice
-
-
-async def async_setup_entry(hass, config_entry, async_add_entities):
-    """Set up the Tesla binary_sensors by config_entry."""
-    entities = [
-        TeslaDeviceEntity(
-            device,
-            hass.data[TESLA_DOMAIN][config_entry.entry_id]["coordinator"],
-        )
-        for device in hass.data[TESLA_DOMAIN][config_entry.entry_id]["devices"][
-            "devices_tracker"
-        ]
-    ]
     async_add_entities(entities, True)
 
 
-class TeslaDeviceEntity(TeslaDevice, TrackerEntity):
-    """A class representing a Tesla device."""
+class CarLocation(TeslaBaseEntity, TrackerEntity):
+    """Representation of the Tesla Car Location Tracker."""
 
-    @property
-    def force_update(self):
-        """Disable forced updated since we are polling via the coordinator updates."""
-        return False
+    def __init__(
+        self, hass: HomeAssistant, car: dict, coordinator: TeslaDataUpdateCoordinator
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        super().__init__(hass, car, coordinator)
+        self.type = "location tracker"
 
     @property
-    def latitude(self) -> float | None:
-        """Return latitude value of the device."""
-        location = self.tesla_device.get_location()
-        return self.tesla_device.get_location().get("latitude") if location else None
+    def source_type(self):
+        """Return the source type, eg gps or router, of the device."""
+        return SOURCE_TYPE_GPS
 
     @property
     def longitude(self) -> float | None:
```

Comment:
Carrying over my comment from the other PR but this is throwing an error when running HA. I don't understand why as this syntax is used elsewhere in the code and Home Assistant. I tried searching around for a solution but couldn't figure it out. Here's the traceback:

````
2022-08-18 02:12:56.306 ERROR (MainThread) [homeassistant.loader] Unexpected exception importing platform custom_components.tesla_custom.device_tracker
Traceback (most recent call last):
  File "/workspaces/home-assistant/homeassistant/loader.py", line 657, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/workspaces/home-assistant/homeassistant/loader.py", line 674, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/workspaces/home-assistant/config/custom_components/tesla_custom/device_tracker.py", line 27, in <module>
    class CarLocation(TeslaCarDevice, TrackerEntity):
  File "/workspaces/home-assistant/config/custom_components/tesla_custom/device_tracker.py", line 43, in CarLocation
    def longitude(self) -> float | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
2022-08-18 02:12:56.308 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform tesla_custom.device_tracker: Platform not found (Exception importing custom_components.tesla_custom.device_tracker).
````

Reply:
I just removed these for now because it's causing the entities to not load. I could not figure out why it's throwing an error since this same syntax and types are used elsewhere in the HA core code.